utils/utils.go

```diff
@@ -134,18 +145,24 @@ func InitPaths() error {
 	return nil
 }
 
-// DownloadFile downloads a file into the cache directory and returns a path to the file.
-func DownloadFile(ctx context.Context, rawURL string, logger logging.Logger) (outPath string, errRet error) {
+// DownloadFile downloads or copies a file into the cache directory and returns a path to the file.
+// If this is an http/s URL, you must check the checksum of the result; the partial logic does not check etags.
+func DownloadFile(ctx context.Context, rawURL string, logger logging.Logger) (string, error) {
 	parsedURL, err := url.Parse(rawURL)
 	if err != nil {
 		return "", err
 	}
+	parsedURL, rewritten := rewriteGCPDownload(parsedURL)
+	if rewritten {
+		logger.Debugf("rewrote GCP URL %q to range-friendly %q", rawURL, parsedURL.String())
```

Comment:
Obligatory "please use structured logging instead of templated log messages" comment. Also same for several other places in this file.

```suggestion
		logger.Debugw("rewrote GCP URL to be range-friendly",
			"rawURL", rawURL, "parsedURL", parsedURL.String())
```

Reply:
ah I've been preferring `f` to `w` bc I thought it was convention in this codebase -- will change

```sh
$ git grep -i logger.infof | wc
     44     345    4757
$ git grep -i logger.infow | wc
     10      73    1119
$ git grep -i logger.debugf | wc
     21     169    2346
$ git grep -i logger.debugw | wc
      3      19     311
```